odl/test/discr/grid_test.py

```diff
@@ -509,6 +507,48 @@ def test_RectGrid_getitem():
     assert grid[1:3] == sub_grid
 
 
+def test_empty_grid():
+    """Check if empty grids behave as expected and all methods work."""
+    grid = RectGrid()
+
+    assert grid.coord_vectors == ()
+    assert grid.ndim == 0
+    assert grid.shape == ()
+    assert grid.size == 0
+    assert len(grid) == 0
+    assert np.array_equal(grid.min_pt, [])
+    assert np.array_equal(grid.max_pt, [])
+    assert np.array_equal(grid.nondegen_byaxis, [])
+    assert grid.is_uniform
+    out = np.array([])
+    grid.min(out=out)
+    grid.max(out=out)
+    assert np.array_equal(grid.mid_pt, [])
+    assert np.array_equal(grid.stride, [])
+    assert np.array_equal(grid.extent, [])
+    assert grid.convex_hull() == odl.IntervalProd([], [])
+    same = RectGrid()
+    assert grid == same
+    other = RectGrid([0, 2, 3])
+    assert grid != other
+    assert grid.is_subgrid(other)
+    assert [] in grid
+    assert 1.0 not in grid
+    assert grid.insert(0, other) == other
+    assert other.insert(0, grid) == other
+    assert other.insert(1, grid) == other
+    assert grid.squeeze() == grid
+    assert np.array_equal(grid.points(), np.array([]).reshape((0, 0)))
+    assert grid.corner_grid() == grid
+    assert np.array_equal(grid.corners(), np.array([]).reshape((0, 0)))
+    assert grid.meshgrid == ()
+    assert grid[[]] == grid
+    assert np.array_equal(np.asarray(grid), np.array([]).reshape((0, 0)))
+    assert grid == uniform_grid([], [], ())
+    hash(grid)
```

Comment:
At least `hash(grid) == hash(same)` and `hash(grid) != hash(other)` when you're at it.